Yosemite/YosemiteTests/Stores/ProductReviewStoreTests.swift

```diff
@@ -123,9 +123,134 @@ final class ProductReviewStoreTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    /// Verifies that `ProductReviewAction.synchronizeProductReviews` for the first page deletes stored Product Reviews for the given site ID.
+    ///
+    func test_syncing_product_reviews_on_the_first_page_resets_stored_product_reviews() {
+
+        // Given
+        let reviewID1: Int64 = 1
+        let reviewID2: Int64 = 2
+        let productReviews = [self.sampleProductReview(reviewID: reviewID1), self.sampleProductReview(reviewID: reviewID2)]
+        self.store.upsertStoredProductReviews(readOnlyProductReviews: productReviews,
+                                              in: self.viewStorage,
+                                              siteID: self.sampleSiteID)
+
+        // When
+        network.simulateResponse(requestUrlSuffix: "products/reviews", filename: "reviews-all")
+        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.ProductReview.self), 2)
+
+        // Then
+        waitForExpectation { exp in
+            let action = ProductReviewAction.synchronizeProductReviews(siteID: sampleSiteID,
+                                                                       pageNumber: defaultPageNumber,
+                                                                       pageSize: defaultPageSize) { error in
+                XCTAssertNil(error)
+
+                // The previously upserted Product Reviews should be deleted.
+                let storedProductReview1 = self.viewStorage.loadProductReview(
+                    siteID: self.sampleSiteID,
+                    reviewID: reviewID1)
+                XCTAssertNil(storedProductReview1)
+
+                let storedProductReview2 = self.viewStorage.loadProductReview(
+                    siteID: self.sampleSiteID,
+                    reviewID: reviewID2)
+                XCTAssertNil(storedProductReview2)
```

Comment:
nit: I think the wait for `ProductReviewAction.synchronizeProductReviews` could be moved to the `// When` section, and then these asserts can be under the `// Then` section?